src/rulesets/Base/nondiff.jl

```diff
@@ -84,6 +84,7 @@
 @non_differentiable Vector(::AbstractArray{Bool})
 
 @non_differentiable Array(::AbstractArray{Bool})
+@non_differentiable BitArray(::Any)
```

Comment:
Is there a reason we didn't splat this?
```julia
@non_differentiable BitArray(::Any...)
```

Reply:
Doesn't it only have 1 constructor?
I didn't actually check

Reply:
```julia
julia> methods(BitArray)
# 6 methods for type constructor:
[1] BitArray(::UndefInitializer, dims::Integer...) in Base at bitarray.jl:69
[2] BitArray(::UndefInitializer, dims::Tuple{Vararg{Integer, N}}) where N in Base at bitarray.jl:71
[3] BitArray(x::BitArray) in Base at bitarray.jl:543
[4] BitArray(A::AbstractArray{var"#s79", N} where var"#s79") where N in Base at bitarray.jl:503
[5] (::Type{T})(x::T) where T<:BitArray in Base at bitarray.jl:542
[6] BitArray(itr) in Base at bitarray.jl:574
```
there seems to be more

Reply:
Should I make another PR?

Reply:
Please do, that would be great, thanks!